### PR TITLE
feat: birthday

### PR DESCRIPTION
**Pull request by flloschy**
Set a Birthday and get Congratulated when the time has come!

### CODE REVIEW

File: src/util/birthdaySystem/loop.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		const adjustedDate = new Date(Date.now() + timezone * 60 * 60 * 1000);
+		const adjustedMonth = adjustedDate.getUTCMonth() + 1;
+		const adjustedDay = adjustedDate.getUTCDate();
+		for (const user of possibleUsers) {
+			const { day, month } = birthdayDb.get(user.id) ?? {};
+			if (!day || !month) continue;
+
+			if (
+				(adjustedMonth === month && adjustedDay === day) ||
+				(month === 1 && day === 29 && adjustedMonth === 2 && adjustedDay === 1)
```
Comment (Chicken) [inline src/util/birthdaySystem/loop.ts:33]: 0-11 vs 1-12
```suggestion
				(month === 2 && day === 29 && adjustedMonth === 3 && adjustedDay === 1)
```

**Review by Chicken — CHANGES_REQUESTED** [review]

File: src/util/birthdaySystem/loop.ts (hunk 1 above)
Comment (flloschy) [reply src/util/birthdaySystem/loop.ts:33]: Isnt this already done with `const adjustedMonth = adjustedDate.getUTCMonth() + 1;` and `const targetHour = config.birthdayNotificationAt - 1;`?

**Review by Chicken — APPROVED** [review]

**Review by StrangeGirlMurph — CHANGES_REQUESTED** [review]
I’ll look at the other things later

File: src/config.ts [hunk 2]
```diff
@@ -61,30 +62,33 @@ export const config: BotConfig = {
...
 	logLevel: "debug",
 	logActivity: true,
-	activityTrackerAdminCommandPermission: PermissionFlagsBits.Administrator,
+	activityTrackerAdminCommandPermission: PermissionsBitField.Flags.Administrator,
+	birthdayNotificationAt: Time.SummerMorning,
 };
 
 interface BotConfig {
-	// Name of the bot.
+	/** Name of the bot. */
```
[the remark this answers is not in the document]
Comment (flloschy) [reply src/config.ts:70]: Before
image

After
image

File: src/util/birthdayCommand/loop.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		}
+	});
+}
+
+function getMembersWithTimeZone(timezone: string, guild: Guild) {
+	return guild.roles.cache.find((role) => role.name == timezone)?.members.map((m) => m);
+}
+
+function getMembersWithNoTimeZone(guild: Guild) {
+	return guild.roles.cache.find((role) => !role.name.startsWith("UTC"))?.members.map((m) => m);
```
Comment (StrangeGirlMurph) [inline src/util/birthdayCommand/loop.ts:56]: this is weird... won't this just return the members of a random role?

**Review by StrangeGirlMurph — CHANGES_REQUESTED** [review]

File: src/util/birthdayCommand/loop.ts (hunk 3 above)
Comment (flloschy) [reply src/util/birthdayCommand/loop.ts:56]: should return ever member who doesn't have a UTC role

Comment (StrangeGirlMurph) [reply src/util/birthdayCommand/loop.ts:56]: nah... find will just return the first element that matches the statement

File: src/events/ready.ts [hunk 4]
```diff
@@ -7,17 +7,21 @@ import {
...
 	logger.info(`Watching over ${client.guilds.cache.size} guilds.`);
 
-	logger.info("Publishing commands...");
-	await updateRegisteredCommands(client).then(() => logger.info("Finished publishing commands."));
-
 	logger.info("Starting reminder scheduler...");
 	ReminderCommand.startScheduler(client);
 
+	logger.info("Starting Birthday scheduler...");
+	BirthdayMessage(client);
```
Comment (StrangeGirlMurph) [inline src/events/ready.ts:20]: Could we keep this consistent with the reminder and have a startScheduler method on the command?

**Review by StrangeGirlMurph — APPROVED** [review]

**Pull request merged**